src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -258,10 +257,10 @@ public void leaveToken(DetailAST ast) {
      */
     private void checkIfSemicolonIsInDifferentLineThanPrevious(DetailAST ast) {
         DetailAST currentStatement = ast;
-        final boolean hasResourcesPrevSibling =
-                currentStatement.getPreviousSibling() != null
-                        && currentStatement.getPreviousSibling().getType() == TokenTypes.RESOURCES;
-        if (!hasResourcesPrevSibling && isMultilineStatement(currentStatement)) {
+        final DetailAST previousSibling = ast.getPreviousSibling();
+        if (previousSibling != null
+            && previousSibling.getType() != TokenTypes.RESOURCES
+            && ast.getParent().getType() != TokenTypes.COMPILATION_UNIT) {
```

Comment:
Something like `changeCurrentStatementToPreviousSibling`?

Reply:
```
final boolean isResourceOrUnnecessarySemi =  previousSibling.getType() == TokenTypes.RESOURCES
            || ast.getParent().getType() == TokenTypes.COMPILATION_UNIT

```

Reply:
Done, thanks for the suggestion.

Reply:
Missed the null check above, name does not really make sense in this context since the AST **would** be a semi after a resource or an extra semi in between imports.

How about:
```
        final boolean isSingleLineStatement = previousSibling == null
            || previousSibling.getType() == TokenTypes.RESOURCES
            || ast.getParent().getType() == TokenTypes.COMPILATION_UNIT;
        if (!isSingleLineStatement) {
            currentStatement = ast.getPreviousSibling();
        }

```
?

Reply:
This condition has nothing to do with the statement being single line or mutliline. How about `isUnnecessarySemicolon`:
```java
final boolean isUnnecessarySemicolon = previousSibling == null
    || previousSibling.getType() == TokenTypes.RESOURCES
    || ast.getParent().getType() == TokenTypes.COMPILATION_UNIT;
if (!isUnnecessarySemicolon) {
    currentStatement = ast.getPreviousSibling();
}
```

Reply:
Yes, this is good too.

Reply:
Done.